src/Chunk.cpp

```diff
@@ -2076,36 +2074,73 @@ bool cChunk::DoWithEntityByID(UInt32 a_EntityID, cLambdaEntityCallback a_Callbac
 
 
 
-bool cChunk::ForEachBlockEntity(cBlockEntityCallback & a_Callback)
+// Helper function - there's probably a better place for this
+
+template <BLOCKTYPE, BLOCKTYPE...> bool IsOneOfImpl(BLOCKTYPE, std::false_type);
+template <class = void> bool IsOneOfImpl(BLOCKTYPE, std::true_type);
```

Comment:
Why exactly do we need the empty arguments? I have done a similar varparam thing with `cLuaState::Push()` and didn't need an empty one.